src/pyqasm/algorithms/solovay_kitaev/optimizer.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+from pyqasm.elements import BasisSet
+
+IDENTITY_WEIGHT_GROUP = {
+    BasisSet.CLIFFORD_T: {
+        "h": {
+            "group": "h",
+            "weight": 0.5
+        },
+        "s": {
+            "group": "s-t",
+            "weight": 0.25
+        },
+        "t": {
+            "group": "s-t",
+            "weight": 0.125
+        },
+        "sdg": {
+            "group": "sdg-tdg",
+            "weight": 0.25
+        },
+        "tdg": {
+            "group": "sdg-tdg",
+            "weight": 0.125
+        },
+    }
+}
+
+def optimize_gate_sequnce(seq: list[str], target_basis_set):
+    target_identity_weight_group = IDENTITY_WEIGHT_GROUP[target_basis_set]
+    while True:
```

Comment:
Let's have a finite limit for this processing, we don't wanna get stuck in the loop. `1e6` iterations should be a good limit to consider